src/main/io/serial_cli.c

```diff
@@ -1710,14 +1710,22 @@ void cliProcess(void)
             clicmd_t target;
             cliPrint("\r\n");
 
-            // Strip trailing whitespace
-            while (bufferIndex > 0 && cliBuffer[bufferIndex - 1] == ' ') {
-                bufferIndex--;
+            // Strip comment starting with # from line
+            char *p = cliBuffer;
+            p = strchr(p, '#');
+            if (NULL != p) {
+                bufferIndex = (uint32_t)(p - cliBuffer);
             }
 
-            cliBuffer[bufferIndex] = 0; // null terminate
+            // Process non-empty lines
+            if (bufferIndex > 0) {
+                // Strip trailing whitespace
+                while (bufferIndex > 0 && cliBuffer[bufferIndex - 1] == ' ') {
```

Comment:
Maybe first remove whitespace, then test if something is left (moving this loop before `if()` )